tritium-caffeine/src/main/java/com/palantir/tritium/metrics/caffeine/CaffeineCacheStats.java

```diff
@@ -90,6 +97,34 @@ public static void registerCache(TaggedMetricRegistry registry, Cache<?, ?> cach
         }
     }
 
+    /**
+     * Creates a {@link StatsCounter} with tagged metrics for tracking caches hits, misses, evictions, and load times.
+     * <p>Example usage:</p>
+     * {@code
+     * Caffeine.newBuilder()
+     *     .recordStats(() -> CaffeineCacheStats.record(taggedMetricRegistry, "name")
```

Comment:
I wonder if we should return a supplier for simplicity?

Reply:
yeah that would make call sites cleaner. will need to `@SuppressWarnings("NoFunctionalReturnType")` though I think that's ok for this specific use

Reply:
We could define an interface which extends `Supplier<StatsCounter>` as well, though that might be less obvious to library consumers.